src/components/ProviderLogin.svelte

```diff
@@ -0,0 +1,174 @@
+    $isLoggedIn = true;
+    pageContent.dashboard();
+  }
+
+  function becomeProvider() {
+    pageContent.becomeProvider();
+  }
+
+  function networkChanged() {
+    console.log('networkChanged');
+    $storeSelectedAccount = '';
+    $transactionSigningAddress = '';
+    canConnect = false;
+  }
+
+  function accountChanged() {
+    console.log('accountChanged');
+    $transactionSigningAddress = $storeSelectedAccount;
+  }
+
+  function isValidURL(url: string): boolean {
+    try {
+      new URL(url);
+      return true;
+    } catch (_) {
+      return false;
+    }
+  }
+
+  function customNetworkChanged(event: KeyboardEvent) {
+    if (event.key === 'Enter') {
+      if (isValidURL(customNetwork)) {
+        const url = new URL(customNetwork);
+        if ($selectedNetwork != null) {
+          $selectedNetwork.endpoint = url;
+        }
+      }
+    }
+  }
+
+  function formatNetwork(network: NetworkInfo): string {
+    if (network.name === 'CUSTOM') {
+      return network.name;
+    }
+    return `${network?.name ?? ''}: ${network?.endpoint?.toString().replace(/\/$/, '') ?? ''}`;
+  }
+</script>
+
+<div id="provider-login" class="content-block flex w-single-block flex-col gap-4">
+  <BlockSection title="Provider Login">
+    <DropDownMenu
+      id="network"
+      label="Select a Network"
+      bind:selected={$selectedNetwork}
+      placeholder="Select a network"
+      options={$allNetworks}
+      onChange={networkChanged}
+      formatter={formatNetwork}
+    />
+    {#if $selectedNetwork != null && $selectedNetwork.name == 'CUSTOM'}
+      <input
+        id="other-endpoint-url"
+        type="text"
+        pattern="^(http:\/\/|https:\/\/|ws:\/\/|wss:\/\/).+"
+        placeholder="wss://some.frequency.node"
+        bind:value={customNetwork}
+        disabled={$selectedNetwork.name != 'CUSTOM'}
+        class:hidden={$selectedNetwork.name != 'CUSTOM'}
+        on:keydown={customNetworkChanged}
+      />
+    {/if}
+    <div id="network-error-msg" class="text-sm text-error">{networkErrorMsg}</div>
+    <DropDownMenu
+      id="controlkeys"
+      label="Select a Provider Control Key"
+      bind:selected={$storeSelectedAccount}
+      placeholder="Select a provider"
+      options={$storeProviderAccounts}
+      onChange={accountChanged}
+      disabled={$storeProviderAccounts.length == 0}
+    />
+    <div id="controlkey-error-msg" class="text-sm text-error">{controlKeysErrorMsg}</div>
+    <Button id="connect-button" title="Connect" disabled={!canConnect} action={connect} />
+  </BlockSection>
+
+  <BlockSection title="Not a Provider?">
+    <div class="text-sm">
+      Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem
+      aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
+    </div>
+    <Button id="request-2b-provider-btn" title="Become a Provider" action={becomeProvider} />
+  </BlockSection>
```

Comment:
Do we want to break this into it's own component too?

Reply:
I don't think so.  There's nothing really to it.